app/locations/controllers.js

```diff
@@ -1,10 +1,28 @@
-const { sortBy } = require('lodash')
+const { get, sortBy } = require('lodash')
 
-function locations(req, res) {
-  const locations = sortBy(req.userLocations, 'title')
+const referenceDataService = require('../../common/services/reference-data')
+
+async function locations(req, res, next) {
+  const userPermissions = get(req.session, 'user.permissions', [])
+
+  let locations = []
+  let regions = []
+
+  if (userPermissions.includes('allocation:create')) {
+    try {
+      regions = await referenceDataService.getRegions()
+      req.session.regions = regions
```

Comment:
Do you need to store the regions on the session? You raised a good point about what needs to happen when serializing and deserializing the session so we can probably drop the need to set it here.

Reply:
Refactored the `setRegion` middleware to _not_ use the values in the session.